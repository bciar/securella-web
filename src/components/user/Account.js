import React from 'react';
import Auth from 'j-toker';
import { RaisedButton, Paper, GridList, GridTile, Avatar } from 'material-ui';

export class Account extends React.Component {
  constructor(props) {
    super(props);
    
    let image_url = Auth.user.image;
    if (!!this.props.location.state) 
        image_url = this.props.location.state.updatedImageURL;
    Auth.user.image = image_url;
  }

  componentWillMount() {
    if (!this.props.user.loggedIn) {
      this.props.history.push('/login');
    }
  }
  
  render() {
    return (
        <div className="container">
            <Paper className="account-paper" zDepth={4}>
                <div id='root'>
                    <div id="first_list">
                      <GridList cols={1} cellHeight={80}>
                          <GridTile rows={1} size={70} className="grid-title">
                            <Avatar size={70} src={ Auth.user.image}  />
                          </GridTile>
                          <GridTile size={30} className="grid-value">
                            <RaisedButton href="/update_image" label="change" />
                          </GridTile>
                      </GridList>
                    </div>
                    <GridList cellHeight={40}>
                        <GridTile className="grid-title"> Name: </GridTile>
                        <GridTile className="grid-value"> {Auth.user.name} </GridTile>
                        <GridTile className="grid-title"> Adresse: </GridTile>
                        <GridTile className="grid-value"> {Auth.user.address} </GridTile>
                        <GridTile className="grid-title"> </GridTile>
                        <GridTile className="grid-value"> {Auth.user.postal_code} {Auth.user.city} </GridTile>
                        <GridTile className="grid-title"> Gender: </GridTile>
                        <GridTile className="grid-value"> {Auth.user.gender} </GridTile>
                        <GridTile className="grid-title"> Age: </GridTile>
                        <GridTile className="grid-value"> {Auth.user.age} </GridTile>
                        <RaisedButton href="/update_profile" label="Update profile" />
                        <RaisedButton href="/reset_password" label="Change password" />
                    </GridList>
                </div>
            </Paper>
        </div>
      );
    }
}
