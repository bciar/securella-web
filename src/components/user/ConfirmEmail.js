import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Paper from 'material-ui/Paper';

export class ConfirmEmail extends Component {

    constructor(props) {
        super(props);
        this.state = {
            confirmed: true,
        };
    }

    render() {
      if (this.state.confirmed) {
        return (
          <div className="container">
            <Paper className="common-paper" zDepth={1}>
              Thank you for confirming your email. You may now proceed to the
              <br/>
              <b><Link to='/login'>Login</Link></b>
            </Paper>
          </div>
        );
      }
      else {
        return (
          <div className="container">
            <Paper className="common-paper" zDepth={1}>
              Sorry, we could not confirm your account.
              <br/>
              <b>Please call 911 for help.</b>
            </Paper>
          </div>
        );
      }
    }
}
