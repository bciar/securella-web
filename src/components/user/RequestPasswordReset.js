import React from 'react';
import Auth from 'j-toker';
import { RaisedButton, Paper } from 'material-ui';
import { ValidatorForm } from 'react-form-validator-core';
import { TextValidator} from 'react-material-ui-form-validator';

export class RequestPasswordReset extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            formData: {
                email: '',
                password: ''
            },
            submitted: false,
            errors: false
        };
    }

    componentWillMount() {
      Auth.validateToken()
        .then(function(user) {
        this.props.history.push('/reset_password');
      });
    }

    handleChange = (event) => {
        const { formData } = this.state;
        formData[event.target.name] = event.target.value;
        this.setState({ formData });
    }

    handleSubmit = (history) => {
        const { email } = this.state.formData;
        this.setState({ submitted: true }, () => {
          Auth.requestPasswordReset({
            email,
            redirect_url: '/reset_password'
          })
          .then(function(user) {
            history.push('/request_password_success');
          })
          .fail(res => {
            document.getElementById("errorLabel").className = '';
            this.setState({ submitted: false });
          })
        });
    }

    render() {
        const { formData, submitted } = this.state;
        if (Auth.user.signedIn) {
            return (<div><h1>You are in!</h1></div>);
        }
        else {
            return (
                <div className="container">
                    <Paper className="common-paper">
                        <ValidatorForm
                            ref="form"
                            onSubmit={ () => this.handleSubmit(this.props.history) } >
                            <h2>Reset your Password</h2>
                            <h3 id="errorLabel" className="hidden"> Email or password wrong</h3>
                            <TextValidator
                                floatingLabelText="Email"
                                onChange={this.handleChange}
                                name="email"
                                id="email"
                                value={formData.email}
                                validators={['required']}
                                errorMessages={['this field is required']}
                            />
                            <br />
                            <RaisedButton
                                type="submit"
                                label={
                                    (submitted && 'Processing...')
                                    || (!submitted && 'Get a new password')
                                }
                                disabled={submitted}
                                style={{marginTop:'50px', marginBottom:'10px'}}
                            />
                        </ValidatorForm>
                    </Paper>
                </div>
            );
        }
    }
}
