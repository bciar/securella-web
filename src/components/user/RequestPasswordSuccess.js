import React from 'react';
import Paper from 'material-ui/Paper';

export const RequestPasswordSuccess = () => (
  <div className="container">
    <Paper className="common-paper" zDepth={1}>
      If your Email is registered, you will receive an email with a link to change your password within the next 15 minutes.
      <br/>
      <b>Please check your inbox.</b>
    </Paper>
  </div>
);