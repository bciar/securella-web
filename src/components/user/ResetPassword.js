import React from 'react';
import Auth from 'j-toker';
import { RaisedButton, Paper } from 'material-ui';
import { ValidatorForm } from 'react-form-validator-core';
import { TextValidator} from 'react-material-ui-form-validator';


export class ResetPassword extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            formData: {
                password: '',
                repeatPassword: '',
            },
            submitted: false,
            errors: '0',
        };
    }

    componentWillMount() {
        ValidatorForm.addValidationRule('isPasswordMatch', (value) => {
            if (value !== this.state.formData.password) {
                return false;
            }
            return true;
        });
    }

    handleChange = (event) => {
        const { formData } = this.state;
        formData[event.target.name] = event.target.value;
        this.setState({ formData });
    }

    handleSubmit = (history) => {
        const { password, repeatPassword } = this.state.formData;
        this.setState({ submitted: true }, () => {
          Auth.updatePassword({
            password: password,
            password_confirmation: repeatPassword
          })
          .then(function() {
            history.push('/reset_password_success');
          })
          .fail(res => {
            document.getElementById("errorLabel").className = '';
            this.setState({ submitted: false });
          })
        });
    }

    render() {
        const { formData, submitted } = this.state;
        return (
            <div className="container">
                <Paper className="common-paper">
                    <ValidatorForm
                        ref="form"
                        onSubmit={ () => this.handleSubmit(this.props.history) } >
                        <h2>Passwort updaten</h2>
                        <h3 id="errorLabel" className="hidden"> Password zu kurz (min. 8 Zeichen)</h3>
                        <TextValidator
                            floatingLabelText="Password"
                            onChange={this.handleChange}
                            name="password"
                            type="password"
                            id="password"
                            validators={['required']}
                            errorMessages={['this field is required']}
                            value={formData.password}
                        />
                        <br />
                        <TextValidator
                            floatingLabelText="Repeat password"
                            onChange={this.handleChange}
                            name="repeatPassword"
                            type="password"
                            id="password_confirmation"
                            validators={['isPasswordMatch', 'required']}
                            errorMessages={['password mismatch', 'this field is required']}
                            value={formData.repeatPassword}
                        />
                        <br />
                        <RaisedButton
                            type="submit"
                            label={
                                (submitted && 'Updating your password!')
                                || (!submitted && 'Passwort updaten')
                            }
                            disabled={submitted}
                            style={{marginTop:'50px', marginBottom:'10px'}}
                        />
                    </ValidatorForm>
                </Paper>
            </div>
        );

    }
}
