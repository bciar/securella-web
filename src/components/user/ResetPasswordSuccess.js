import React from 'react';
import { Link } from 'react-router-dom';
import Paper from 'material-ui/Paper';

export const ResetPasswordSuccess = () => (
  <div className="container">
    <Paper className="common-paper" zDepth={1}>
      <h1>You are updated your password</h1>
      <p>
        <Link to='/login'>Login</Link>&nbsp;&nbsp;here now.
      </p>
    </Paper>
  </div>
);