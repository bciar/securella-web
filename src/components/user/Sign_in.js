import React from 'react';
import Auth from 'j-toker';
import { Toggle, RaisedButton } from 'material-ui';
import { ValidatorForm } from 'react-form-validator-core';
import { TextValidator} from 'react-material-ui-form-validator';
import { Link, Redirect } from 'react-router-dom';
import store from '../../store.js';
import { setLoginState } from '../../actions/userActions.js'

const submitButtonStyle = {
    background: '#49ca89',
    borderRadius: '50%',
    width: '50px',
    height: '50px',
    minWidth: '50px',
};

export class SignIn extends React.Component {
    constructor(props) {
        super(props);
        const email = localStorage.getItem('email') || '';
        const password = localStorage.getItem('password') || '';
        this.state = {
            formData: {
                email,
                password
            },
            submitted: false,
            errors: false, 
            isSaveLogin: true,
        };
    }

    componentWillMount() {
      if (this.props.user.loggedIn) {
        this.props.history.push('/dashboard');
      }
    }

    handleChange = (event) => {
        const { formData } = this.state;
        formData[event.target.name] = event.target.value;
        this.setState({ formData });
    }

    handleSubmit = (history) => {
        const state = this.state;
        const {email, password} = state.formData;
        this.setState({ submitted: true }, () => {
            Auth.emailSignIn({
                email: email,
                password: password,
            })
            .then(function(user) {
                if (state.isSaveLogin) {
                    localStorage.setItem('email', email);
                    localStorage.setItem('password', password);
                }
                else {
                    localStorage.removeItem('email');
                    localStorage.removeItem('password');
                }
                store.dispatch(setLoginState(true));
            })
            .fail(res => {
                document.getElementById("errorLabel").className ='';
                this.setState({ submitted: false });
            })
        });
    }

    render() {
        const { formData, submitted } = this.state;
        if (this.props.user.loggedIn) { return <Redirect to='/dashboard' />; }
        else {
            return (
                <ValidatorForm
                    ref="form"
                    name="signinform"
                    className="signin-form"
                    autoComplete="off"
                    onSubmit={ () => this.handleSubmit(this.props.history) } 
                >
                    <div className="login-wrapper">
                        <h2>Securella Login</h2>
                        <h3 id="errorLabel" className="hidden"> Email or password wrong</h3>
                        <TextValidator
                            floatingLabelText="Email"
                            onChange={this.handleChange}
                            name="email"
                            id="email"
                            value={formData.email}
                            validators={['required']}
                            errorMessages={['this field is required']}
                        />
                        <br />
                        <TextValidator
                            floatingLabelText="Password"
                            onChange={this.handleChange}
                            name="password"
                            type="password"
                            id="password"
                            value={formData.password}
                            validators={['required']}
                            errorMessages={['this field is required']}
                        />
                        <div className="buttons-container">
                            <div className="buttons">
                                <Toggle
                                    className="toggle"
                                    label="Save login"
                                    defaultToggled={true}
                                    onToggle={(e, value) => this.setState({ isSaveLogin: value })}
                                />
                                <RaisedButton
                                    type="submit"
                                    className="btn-ok"
                                    style={submitButtonStyle}
                                    label={submitted ? '...' : 'OK'}
                                    disabled={ submitted }
                                />
                            </div>
                            <p><Link to='/request_password_reset'>forget your password ?</Link></p>
                        </div>
                    </div>
                </ValidatorForm>
            );
        }
    }
}
