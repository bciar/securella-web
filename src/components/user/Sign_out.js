import React from 'react';
import Auth from 'j-toker';
import Paper from 'material-ui/Paper';
import { Link } from 'react-router-dom';
import store from '../../store.js';
import {setLoginState} from '../../actions/userActions.js'

export class SignOut extends React.Component {
    componentWillMount() {
        Auth.signOut();
        this.props.history.push('/');
        store.dispatch(setLoginState(false));
    }
    render() {
        return (
            <div className="container">
                <Paper className="common-paper" zDepth={1}>
                  <h1>You are logged out</h1>
                  <p>
                    <Link to='/login'>Login</Link> here if you want to go back.
                  </p>
                </Paper>
            </div>
        );
    }
}
