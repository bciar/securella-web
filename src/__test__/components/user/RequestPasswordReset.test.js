import React from 'react';
import { shallow } from 'enzyme';
import { RequestPasswordReset } from '../../../components/user/RequestPasswordReset';

let history, wrapper;
beforeEach(() => {
    history = { push: jest.fn() };
    wrapper = shallow(<RequestPasswordReset history={history} />);
});

test('RequestPasswordReset page should be rendered', () => {
    expect(wrapper).toMatchSnapshot();
});

test('State should be changed correctly according to the user input', () => {
    const test_email = 'test-user@mail.com';
    wrapper.find('TextValidator').simulate('change', {
        target: { name: 'email', value: test_email }
    });
    expect(wrapper.state().formData.email).toEqual(test_email);
});

test('State submitted should be true', () => {
    wrapper.find('ValidatorForm').simulate('submit', history);
    expect(wrapper.state().submitted).toEqual(true);
})