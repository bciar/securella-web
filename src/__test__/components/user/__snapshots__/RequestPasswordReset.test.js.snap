// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestPasswordReset page should be rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RequestPasswordReset
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Paper
        circle={false}
        className="common-paper"
        rounded={true}
        transitionEnabled={true}
        zDepth={1}
>
        <ValidatorForm
                onError={[Function]}
                onSubmit={[Function]}
        >
                <h2>
                        Reset your Password
                </h2>
                <h3
                        className="hidden"
                        id="errorLabel"
                >
                         Email or password wrong
                </h3>
                <TextValidator
                        errorMessages={
                                Array [
                                        "this field is required",
                                      ]
                        }
                        floatingLabelText="Email"
                        id="email"
                        name="email"
                        onChange={[Function]}
                        validatorListener={[Function]}
                        validators={
                                Array [
                                        "required",
                                      ]
                        }
                        value=""
                />
                <br />
                <RaisedButton
                        disabled={false}
                        fullWidth={false}
                        label="Get a new password"
                        labelPosition="after"
                        primary={false}
                        secondary={false}
                        style={
                                Object {
                                        "marginBottom": "10px",
                                        "marginTop": "50px",
                                      }
                        }
                        type="submit"
                />
        </ValidatorForm>
</Paper>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ValidatorForm
          onError={[Function]}
          onSubmit={[Function]}
>
          <h2>
                    Reset your Password
          </h2>
          <h3
                    className="hidden"
                    id="errorLabel"
          >
                     Email or password wrong
          </h3>
          <TextValidator
                    errorMessages={
                              Array [
                                        "this field is required",
                                      ]
                    }
                    floatingLabelText="Email"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    validatorListener={[Function]}
                    validators={
                              Array [
                                        "required",
                                      ]
                    }
                    value=""
          />
          <br />
          <RaisedButton
                    disabled={false}
                    fullWidth={false}
                    label="Get a new password"
                    labelPosition="after"
                    primary={false}
                    secondary={false}
                    style={
                              Object {
                                        "marginBottom": "10px",
                                        "marginTop": "50px",
                                      }
                    }
                    type="submit"
          />
</ValidatorForm>,
        "circle": false,
        "className": "common-paper",
        "rounded": true,
        "transitionEnabled": true,
        "zDepth": 1,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <h2>
              Reset your Password
</h2>,
            <h3
              className="hidden"
              id="errorLabel"
>
               Email or password wrong
</h3>,
            <TextValidator
              errorMessages={
                            Array [
                                          "this field is required",
                                        ]
              }
              floatingLabelText="Email"
              id="email"
              name="email"
              onChange={[Function]}
              validatorListener={[Function]}
              validators={
                            Array [
                                          "required",
                                        ]
              }
              value=""
/>,
            <br />,
            <RaisedButton
              disabled={false}
              fullWidth={false}
              label="Get a new password"
              labelPosition="after"
              primary={false}
              secondary={false}
              style={
                            Object {
                                          "marginBottom": "10px",
                                          "marginTop": "50px",
                                        }
              }
              type="submit"
/>,
          ],
          "onError": [Function],
          "onSubmit": [Function],
        },
        "ref": "form",
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Reset your Password",
            },
            "ref": null,
            "rendered": "Reset your Password",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Email or password wrong",
              "className": "hidden",
              "id": "errorLabel",
            },
            "ref": null,
            "rendered": " Email or password wrong",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "errorMessages": Array [
                "this field is required",
              ],
              "floatingLabelText": "Email",
              "id": "email",
              "name": "email",
              "onChange": [Function],
              "validatorListener": [Function],
              "validators": Array [
                "required",
              ],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "fullWidth": false,
              "label": "Get a new password",
              "labelPosition": "after",
              "primary": false,
              "secondary": false,
              "style": Object {
                "marginBottom": "10px",
                "marginTop": "50px",
              },
              "type": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Paper
          circle={false}
          className="common-paper"
          rounded={true}
          transitionEnabled={true}
          zDepth={1}
>
          <ValidatorForm
                    onError={[Function]}
                    onSubmit={[Function]}
          >
                    <h2>
                              Reset your Password
                    </h2>
                    <h3
                              className="hidden"
                              id="errorLabel"
                    >
                               Email or password wrong
                    </h3>
                    <TextValidator
                              errorMessages={
                                        Array [
                                                  "this field is required",
                                                ]
                              }
                              floatingLabelText="Email"
                              id="email"
                              name="email"
                              onChange={[Function]}
                              validatorListener={[Function]}
                              validators={
                                        Array [
                                                  "required",
                                                ]
                              }
                              value=""
                    />
                    <br />
                    <RaisedButton
                              disabled={false}
                              fullWidth={false}
                              label="Get a new password"
                              labelPosition="after"
                              primary={false}
                              secondary={false}
                              style={
                                        Object {
                                                  "marginBottom": "10px",
                                                  "marginTop": "50px",
                                                }
                              }
                              type="submit"
                    />
          </ValidatorForm>
</Paper>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ValidatorForm
            onError={[Function]}
            onSubmit={[Function]}
>
            <h2>
                        Reset your Password
            </h2>
            <h3
                        className="hidden"
                        id="errorLabel"
            >
                         Email or password wrong
            </h3>
            <TextValidator
                        errorMessages={
                                    Array [
                                                "this field is required",
                                              ]
                        }
                        floatingLabelText="Email"
                        id="email"
                        name="email"
                        onChange={[Function]}
                        validatorListener={[Function]}
                        validators={
                                    Array [
                                                "required",
                                              ]
                        }
                        value=""
            />
            <br />
            <RaisedButton
                        disabled={false}
                        fullWidth={false}
                        label="Get a new password"
                        labelPosition="after"
                        primary={false}
                        secondary={false}
                        style={
                                    Object {
                                                "marginBottom": "10px",
                                                "marginTop": "50px",
                                              }
                        }
                        type="submit"
            />
</ValidatorForm>,
          "circle": false,
          "className": "common-paper",
          "rounded": true,
          "transitionEnabled": true,
          "zDepth": 1,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h2>
                Reset your Password
</h2>,
              <h3
                className="hidden"
                id="errorLabel"
>
                 Email or password wrong
</h3>,
              <TextValidator
                errorMessages={
                                Array [
                                                "this field is required",
                                              ]
                }
                floatingLabelText="Email"
                id="email"
                name="email"
                onChange={[Function]}
                validatorListener={[Function]}
                validators={
                                Array [
                                                "required",
                                              ]
                }
                value=""
/>,
              <br />,
              <RaisedButton
                disabled={false}
                fullWidth={false}
                label="Get a new password"
                labelPosition="after"
                primary={false}
                secondary={false}
                style={
                                Object {
                                                "marginBottom": "10px",
                                                "marginTop": "50px",
                                              }
                }
                type="submit"
/>,
            ],
            "onError": [Function],
            "onSubmit": [Function],
          },
          "ref": "form",
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Reset your Password",
              },
              "ref": null,
              "rendered": "Reset your Password",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Email or password wrong",
                "className": "hidden",
                "id": "errorLabel",
              },
              "ref": null,
              "rendered": " Email or password wrong",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "errorMessages": Array [
                  "this field is required",
                ],
                "floatingLabelText": "Email",
                "id": "email",
                "name": "email",
                "onChange": [Function],
                "validatorListener": [Function],
                "validators": Array [
                  "required",
                ],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "disabled": false,
                "fullWidth": false,
                "label": "Get a new password",
                "labelPosition": "after",
                "primary": false,
                "secondary": false,
                "style": Object {
                  "marginBottom": "10px",
                  "marginTop": "50px",
                },
                "type": "submit",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
